### PR TITLE
DSPDC-530 Add root elements to outputs.

### PR DESCRIPTION
**Pull request by danxmoran**
Logic was getting hairy, so added a bunch of comments too.

### CODE REVIEW

File: xml/src/main/scala/org/broadinstitute/monster/extractors/xml/XmlExtractor.scala [hunk 1]
```diff
@@ -89,37 +141,66 @@ class XmlExtractor[Path] private[xml] (
...
-              }.flatMap { _ =>
-                IO.delay(outputStream.write('\n'))
+    // Pull the stream until we find the first start element, which is the document root.
+    xmlEventStream
+      .dropWhile(!_.isStartElement)
+      .pull
+      .uncons1
+      .flatMap {
+        // No root found (document is empty?)
+        case None => Pull.done
```
Comment (benjamincarlin) [inline xml/src/main/scala/org/broadinstitute/monster/extractors/xml/XmlExtractor.scala:154]: Do we have a test for this?

**Review by benjamincarlin — COMMENTED** [review]
The comments for the recursive portion of the PR are really nice! Great work!!!!!!

File: xml/src/main/scala/org/broadinstitute/monster/extractors/xml/XmlExtractor.scala (hunk 1 above)
Comment (danxmoran) [reply xml/src/main/scala/org/broadinstitute/monster/extractors/xml/XmlExtractor.scala:154]: We do now 😎 Nice catch, there was a bug.

File: xml/src/test/scala/org/broadinstitute/monster/extractors/xml/XmlExtractorSpec.scala [hunk 2]
```diff
@@ -42,15 +42,19 @@ class XmlExtractorSpec extends FlatSpec with Matchers with EitherValues {
...
-      val expectedJsons = readLocal(out)
-        .through(fs2.text.utf8Decode)
-        .through(fs2.text.lines)
-        .filter(!_.isEmpty)
-        .map(io.circe.parser.parse)
-        .rethrow
-        .compile
-        .toList
-        .unsafeRunSync()
+      val expectedJsons = if (buf.isEmpty) {
```
Comment (benjamincarlin) [inline xml/src/test/scala/org/broadinstitute/monster/extractors/xml/XmlExtractorSpec.scala:45]: Nice!

**Review by benjamincarlin — APPROVED** [review]

File: project/build.properties [hunk 3]
```diff
@@ -1 +1 @@
-sbt.version=1.3.0-RC2
+sbt.version=1.3.0
```
Comment (raaidbroad) [inline project/build.properties:1]: Not really a concern, but just curious: is there any downside to our projects having different sbt versions?

**Review by raaidbroad — APPROVED** [review]

File: project/build.properties (hunk 3 above)
Comment (danxmoran) [reply project/build.properties:1]: No downside in terms of technical cross-compatibility, but potential downside of introducing more context-switches to our dev flow if they drift too far apart.

**Pull request merged**